pkg/driver/node/mounter/systemd_mounter.go

```diff
@@ -168,48 +164,10 @@ func (m *SystemdMounter) Mount(bucketName string, target string, credentials *Mo
 	return nil
 }
 
-// Moves a parameter optionName from the options list to MP's environment variable list. We need this as options are
-// passed to the driver in a single field, but MP sometimes only supports config from environment variables.
-// Returns an updated options and environment.
-func moveOptionToEnvironmentVariables(optionName string, envName string, options []string, env []string) ([]string, []string) {
-	optionIdx := -1
-	for i, o := range options {
-		if strings.HasPrefix(o, optionName) {
-			optionIdx = i
-			break
-		}
-	}
-	if optionIdx != -1 {
-		// We can do replace here as we've just verified it has the right prefix
-		env = append(env, strings.Replace(options[optionIdx], optionName, envName, 1))
-		options = append(options[:optionIdx], options[optionIdx+1:]...)
-	}
-	return options, env
-}
-
-// method to add the user agent prefix to the Mountpoint headers
-// https://github.com/awslabs/mountpoint-s3/pull/548
-func addUserAgentToOptions(options []string, userAgent string) []string {
-	// first remove it from the options in case it's in there
-	for i := len(options) - 1; i >= 0; i-- {
-		if strings.Contains(options[i], userAgentPrefix) {
-			options = append(options[:i], options[i+1:]...)
-		}
-	}
-	// add the hard coded S3 CSI driver user agent string
-	return append(options, userAgentPrefix+"="+userAgent)
-}
```

Comment:
The logic for moving `--aws-max-attempts` to env and adding user-agent to arguments has been moved to `node.go` as it's unrelated to mount mechanism as needs to be there for both `SystemdMounter` and `PodMounter`.